### PR TITLE
CIF-2618: make StoreConfigExporter a ConsumerType

### PR DESCRIPTION
**Pull request by Buuhuu**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/client/MagentoGraphqlClient.java [hunk 1]
```diff
@@ -81,6 +81,15 @@
...
+    @Deprecated
     Map<String, String> getHttpHeaders();
+
+    /**
+     * Returns the Map of custom HTTP headers used by the GraphQL client.
+     * 
+     * @return a {@link Map} with header names as keys and header values as values
+     */
+    Map<String, String[]> getHttpHeaderMap();
 }
```
Comment (mhaack) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/client/MagentoGraphqlClient.java:95]: While I don't disagree with this improvement, how is that related to the PR goal of making StoreConfigExporter a ConsumerType?

**Review by mhaack — APPROVED** [review]

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/client/MagentoGraphqlClient.java (hunk 1 above)
Comment (Buuhuu) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/client/MagentoGraphqlClient.java:95]: It is related to the change in the StoreConfigExporter interface. 

Initially we used the `StoreConfigExporterImpl` in the `PageImpl` because we wanted to have a method that exposes the http headers as Map instead of an already serialised object. This did not require a change to the interface at this time.

However, now that we use the interface we had to add this method to the interface as well and as headers can appear more than once using a `Map<String, String[]>` is the right return value for that. (See also the servlet spec). As the Map of headers returned by the StoreConfigExporter is directly derived from the MagentoGraphqlClient, I created the method that returns the data in the same, correct format.

Bottom line: this has been a design mistake in the first place and this is an attempt to clean things up in as less breaking iterations as possible.

Comment (mhaack) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/client/MagentoGraphqlClient.java:95]: > Bottom line: this has been a design mistake in the first place and this is an attempt to clean things up in as less breaking iterations as possible.

That was my thinking as well for question. :-)
